test/lib/ufe/testDeleteCmd.py

```diff
@@ -310,10 +421,240 @@ def testDeleteArgs(self):
+        ufe.Scene.addObserver(ufeObs)
+
+        # clear selection to start off
+        cmds.select(clear=True)
+        
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|Asset_flattened_instancing_and_class_removed_usd|Asset_flattened_instancing_and_class_removed_usdShape')
+        usdPathSegment = usdUtils.createUfePathSegment('/apple/payload/geo')
+        geoPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        geoItem = ufe.Hierarchy.createItem(geoPath)
+        globalSn = ufe.GlobalSelection.get()        
+        globalSn.append(geoItem)
+        self.assertEqual(len(ufe.GlobalSelection.get()), 1)
+
+        cmds.delete()
+        # not deleted
+        self.assertEqual(ufeObs.nbDeleteNotif(), 0)
+
+        # Clear the selection.
+
+        globalSn.clear()
+        self.assertTrue(globalSn.empty())
 
+        usdPathSegment = usdUtils.createUfePathSegment('/apple/payload')
+        geoPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        geoItem = ufe.Hierarchy.createItem(geoPath)
+
+        globalSn.append(geoItem)
+        self.assertEqual(len(ufe.GlobalSelection.get()), 1)
+
+        cmds.delete()
+        #deleted
+        self.assertEqual(ufeObs.nbDeleteNotif(), 1)
+
+    @unittest.skipUnless(ufeUtils.ufeFeatureSetVersion() >= 2, 'testDeleteProxyShape only available in UFE v2 or greater.') 
+    def testDeleteUniqueName(self):
+
+        cmds.file(new=True, force=True)
+
+        # open tree.ma scene in testSamples
+        mayaUtils.openTreeScene()
+		
+        # Create some extra Maya nodes
+        cmds.polySphere()
+		
+        # clear selection to start off
+        cmds.select(clear=True)
+
+        # Create our UFE notification observer
+        ufeObs = TestObserver()
+        ufe.Scene.addObserver(ufeObs)
+		
+        # select a maya object.
+        spherePath = ufe.Path(mayaUtils.createUfePathSegment("|pSphere1"))
+        sphereItem = ufe.Hierarchy.createItem(spherePath)
+        sphereShapePath = ufe.Path(
+            mayaUtils.createUfePathSegment("|pSphere1|pSphereShape1"))
+        sphereShapeItem = ufe.Hierarchy.createItem(sphereShapePath)
+
+        ufe.GlobalSelection.get().append(sphereShapeItem)
+
+        # select a USD object.
+        mayaPathSegment = mayaUtils.createUfePathSegment('|Tree_usd|Tree_usdShape')
+        usdPathSegment = usdUtils.createUfePathSegment('/TreeBase/trunk')
+        trunkPath = ufe.Path([mayaPathSegment, usdPathSegment])
+        trunkItem = ufe.Hierarchy.createItem(trunkPath)
+
+        ufe.GlobalSelection.get().append(trunkItem)
+
+        # get the USD stage
+        stage = mayaUsd.ufe.getStage(str(mayaPathSegment))
+
+        # by default edit target is set to the Rootlayer.
+        self.assertEqual(stage.GetEditTarget().GetLayer(), stage.GetRootLayer())
+
+        # delete selected item `/TreeBase/trunk` `
+        cmds.delete()
+        # deleted two items
+        self.assertEqual(ufeObs.nbDeleteNotif() , 2)
```

Comment:
Again I'd check beyond the number of notifications.  Also, this test feels like there's quite a bit of overlap with the last part of the testDeleteRestrictionHasSpecs test, unless I'm missing something --- if I am, might be good to add comments about what the difference is.